### PR TITLE
PKX-465 PKX-467 Added featured course, course set slider and course library

### PR DESCRIPTION
**Pull request by mubtadaali**
### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: openedx/features/pakx/lms/overrides/utils.py [hunk 1]
```diff
@@ -33,16 +35,64 @@
...
+    """
+    Get course data required for home page course card
+
+    :returns (dict): dict of course card data
+    """
+    pakx_short_logo = '/static/pakx/images/mooc/pX.png'
+    course_custom_setting = getattr(course, 'custom_settings', None)
+
+    if not org_prefetched:
+        course_org = Organization.objects.filter(name__iexact=course.org).first()
```
Comment (emzubair) [inline openedx/features/pakx/lms/overrides/utils.py:48]: shouldn't we use org's `short_name` here, usually courses org are like pakx, btc, ttc

File: openedx/features/pakx/lms/overrides/utils.py [hunk 2]
```diff
@@ -33,16 +35,64 @@
...
 
-def get_featured_course():
+def get_course_card_data(course, org_prefetched=False):
+    """
+    Get course data required for home page course card
+
+    :returns (dict): dict of course card data
+    """
+    pakx_short_logo = '/static/pakx/images/mooc/pX.png'
+    course_custom_setting = getattr(course, 'custom_settings', None)
```
Comment (emzubair) [inline openedx/features/pakx/lms/overrides/utils.py:45]: Suggestion: we should write a util function, which takes course & performs get_or_create functionality for custom_settings
WDYT?
This way we can reduce custom_settings null checks on multiple places.

File: openedx/features/pakx/lms/overrides/utils.py [hunk 3]
```diff
@@ -1,16 +1,18 @@
 """ Overrides app util functions """
 
+import re
 from datetime import datetime
 from logging import getLogger
-from re import compile, findall
```
Comment (emzubair) [inline openedx/features/pakx/lms/overrides/utils.py:5]: Why did we revert this?

**Review by emzubair — CHANGES_REQUESTED** [review]

File: openedx/features/pakx/lms/overrides/utils.py (hunk 3 above)
Comment (mubtadaali) [reply openedx/features/pakx/lms/overrides/utils.py:5]: conflicts with python's built-in function name, plus clear indication of using the re module for better readbility.

File: openedx/features/pakx/cms/custom_settings/views.py [hunk 4]
```diff
@@ -59,12 +61,23 @@ def post(self, request, course_key_string):
...
         course_key = CourseKey.from_string(course_key_string)
+
+        course_set = request.POST['course-set']
         course_overview = request.POST['course-overview']
+        card_description = request.POST['card-description']
+        publisher_logo_url = request.POST['publisher-logo-url']
         course_experience = request.POST.get('course_experience', 0)
+
         if course_overview is not None:
             CourseOverviewContent.objects.update_or_create(
```
Comment (emzubair) [inline openedx/features/pakx/cms/custom_settings/views.py:72]: shouldn't we call `get_or_create_course_overview_content` here?

File: openedx/features/pakx/lms/overrides/utils.py [hunk 5]
```diff
@@ -34,6 +37,72 @@
...
+        'url': reverse('about_course', kwargs={'course_id': text_type(course.id)}),
+    }
+
+
+def get_featured_course_set():
+    """
+    Get featured course-set data in a list of dict
+    :return (list): return list of dict containing course data
+    """
+    feature_course_set_key = configuration_helpers.get_value('feature_course_set_name')
```
Comment (emzubair) [inline openedx/features/pakx/lms/overrides/utils.py:82]: Suggestion: `feature_course_set_key` should be named `feature_course_set_name`, current name creates confusion in the following expression
`custom_settings__course_set__name__iexact=feature_course_set_key`

**Review by emzubair — CHANGES_REQUESTED** [review]

File: openedx/features/pakx/cms/custom_settings/views.py (hunk 4 above)
Comment (mubtadaali) [reply openedx/features/pakx/cms/custom_settings/views.py:72]: get_or_create_course_overview_content does not update the record. can't use that function here.

**Review by emzubair — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**